File: lezlyM/Python/Enteros/Enteros.py
def clasificar_numeros(n):
    pares = []
    impares = []
    primos = []

    for num in range(1, n + 1):
        # Clasificación de Pares e Impares
        if num % 2 == 0:
            pares.append(num)
        else:
            impares.append(num)

        # Clasificación de Primos
        if num > 1:
            es_primo = True
            for i in range(2, int(num ** 0.5) + 1):
                if num % i == 0:
                    es_primo = False
                    break
            if es_primo:
                primos.append(num)

    # Resultados
    print(f"--- Resultados hasta {n} ---")
    print(f"Pares ({len(pares)}): {pares}")
    print(f"Impares ({len(impares)}): {impares}")
    print(f"Primos ({len(primos)}): {primos}")


# Ejemplo de uso con n = 20
clasificar_numeros(20)

File: lezlyM/Rust/Enteros/enteros.rs
use std::io;

fn main() {

    // pedir número al usuario
    println!("77:");

    let mut entrada = String::new();
    io::stdin()
        .read_line(&mut entrada)
        .expect("Error al leer");

    let n: i32 = entrada.trim().parse().expect("54");

    // lista de números
    let mut num: Vec<i32> = Vec::new();

    for i in 0..=n {
        num.push(i);
    }

    // listas de clasificación
    let mut primo: Vec<i32> = Vec::new();
    let mut pares: Vec<i32> = Vec::new();
    let mut impares: Vec<i32> = Vec::new();

    for i in &num {

        // clasificación pares e impares
        if i % 2 == 0 {
            pares.push(*i);
        } else {
            impares.push(*i);
        }

        // verificación de primos
        if *i > 1 {

            let mut es_primo = true;

            for j in 2..*i {
                if i % j == 0 {
                    es_primo = false;
                    break;
                }
            }

            if es_primo {
                primo.push(*i);
            }
        }
    }

    // impresión de resultados
    println!("Numeros pares: {:?}", pares);
    println!("Cantidad de pares: {}", pares.len());

    println!("Numeros impares: {:?}", impares);
    println!("Cantidad de impares: {}", impares.len());

    println!("Numeros primos: {:?}", primo);
    println!("Cantidad de primos: {}", primo.len());
}
